lib/api/controllers/adminController.js

```diff
@@ -51,6 +53,107 @@ function AdminController(kuzzle) {
   };
 
   /**
+   * Get the collection validation specifications
+   * @param {RequestObject} requestObject
+   * @returns {Promise}
+   */
+  this.getSpecifications = requestObject => {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('data:beforeGetSpecifications', requestObject)
+      .then(newRequestObject => {
+        modifiedRequestObject = newRequestObject;
+        return internalEngine.get('validations', `${modifiedRequestObject.index}#${modifiedRequestObject.collection}`);
+      })
+      .then(response => kuzzle.pluginsManager.trigger('data:afterGetSpecifications', new ResponseObject(modifiedRequestObject, response._source)));
+  };
+
+  /**
+   * Add a specification collections
+   * @param {RequestObject} requestObject
+   * @returns {Promise}
+   */
+  this.updateSpecifications = requestObject => {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('data:beforeUpdateSpecifications', requestObject)
+      .then(newRequestObject => {
+        modifiedRequestObject = newRequestObject;
+
+        return prepareSpecificationValidation(modifiedRequestObject, kuzzle);
+      })
+      .then(response => {
+        var promises = [];
+        if (response.error) {
+          kuzzle.pluginsManager.trigger('log:error', `Validation specification has not been updated because of an error: ${requestObject.data.body}.`);
+          return Promise.resolve(response.responseObject);
+        }
+        response.specifications.forEach(specification => {
+          var specLogName = specification._id.split('#').join(' / ');
+          promises.push(internalEngine.createOrReplace('validations', specification._id, specification._source));
+          kuzzle.pluginsManager.trigger('log:info', `Validation specification for ${specLogName} is about to be stored.`);
+        });
+
+        return Promise.all(promises)
```

Comment:
... and you understood it :) (the reason why I did not reject the promise)

You may be right